scout/server/blueprints/variant/utils.py

```diff
@@ -0,0 +1,453 @@
+                   case_obj['individuals']}
+    for sample_obj in variant_obj['samples']:
+        individual = individuals[sample_obj.get('sample_id')]
+        if not individual:
+            continue
+        sample_obj['is_affected'] = False 
+        
+        if individual['phenotype'] == 2:
+            sample_obj['is_affected'] = True 
+
+def evaluation(store, evaluation_obj):
+    """Fetch and fill-in evaluation object."""
+    evaluation_obj['institute'] = store.institute(evaluation_obj['institute_id'])
+    evaluation_obj['case'] = store.case(evaluation_obj['case_id'])
+    evaluation_obj['variant'] = store.variant(evaluation_obj['variant_specific'])
+    evaluation_obj['criteria'] = {criterion['term']: criterion for criterion in
+                                  evaluation_obj['criteria']}
+    evaluation_obj['classification'] = ACMG_COMPLETE_MAP.get(evaluation_obj['classification'])
+    return evaluation_obj
+
+def transcript_str(transcript_obj, gene_name=None):
+    """Generate amino acid change as a string.
+    
+    Args:
+        transcript_obj(dict)
+        gene_name(str)
+    
+    Returns:
+        change_str(str): A description of the transcript level change
+    """
+    # variant between genes
+    gene_part = 'intergenic'
+    part_count_raw = '0'
+
+    if transcript_obj.get('exon'):
+        gene_part = 'exon'
+        part_count_raw = transcript_obj['exon']
+    elif transcript_obj.get('intron'):
+        gene_part = 'intron'
+        part_count_raw = transcript_obj['intron']
+
+    part_count = part_count_raw.rpartition('/')[0]
+    change_str = "{}:{}{}:{}:{}".format(
+        transcript_obj.get('refseq_id', ''),
+        gene_part,
+        part_count,
+        transcript_obj.get('coding_sequence_name', 'NA'),
+        transcript_obj.get('protein_sequence_name', 'NA'),
+    )
+    if gene_name:
+        change_str = "{}:".format(gene_name) + change_str
+
+    return change_str
+
+
+def frequency(variant_obj):
+    """Returns a judgement on the overall frequency of the variant.
+
+    Combines multiple metrics into a single call.
+    
+    Args:
+        variant_obj(scout.models.Variant)
+    
+    Returns:
+        str in ['common','uncommon','rare']
+    """
+    most_common_frequency = max(variant_obj.get('thousand_genomes_frequency') or 0,
+                                variant_obj.get('exac_frequency') or 0,
+                                variant_obj.get('gnomad_frequency') or 0
+                                )
+
+    if most_common_frequency > .05:
+        return 'common'
+    if most_common_frequency > .01:
+        return 'uncommon'
+    
+    return 'rare'
+
+
+def end_position(variant_obj):
+    """Calculate end position for a variant.
+    
+    Args:
+        variant_obj(scout.models.Variant)
+    
+    Returns:
+        end_position(int)
+    """
+    alt_bases = len(variant_obj['alternative'])
+    num_bases = max(len(variant_obj['reference']), alt_bases)
```

Comment:
No, can't follow. What if both len(ref) > 1 and len(alt)> 1 as in a classical indel?

Reply:
Also not compatible with SVs obviously, but I suppose its never called for them.. 😊

Reply:
This is not used for SVs. Since all variants are normalised when loaded that situation should not happen

Reply:
I may just be slow: take it step wise! Normalise would to me only mean the alt and ref are as short as can be, and correctly adjusted to one side? C f https://genome.sph.umich.edu/wiki/Variant_Normalization; the first blue CAT->TGC is normalised but both alleles have size >1.

Reply:
Aha.. and in that case one would like to have length 3?

Reply:
Right; which still works I guess? And the next step would be to imagine two alleles with different >1 lengths...

But more to the point is perhaps that I don't exactly see why the variant end position would not always just be the variant start pos plus length of ref allele? Even if the alt allele is a km long, I right now don't see how that could affect the position on the reference?